core/include/userver/engine/io/tls_wrapper.hpp

```diff
@@ -87,6 +92,11 @@ class USERVER_NODISCARD TlsWrapper final : public ReadableBase {
     return RecvAll(buf, len, deadline);
   }
 
+  /// @brief marks socket as "not-awaitable": no coroutine-related waiting
```

Comment:
`engine::io` is not intended for usage outside of coroutine environment and should stay that way.

Reply:
Could you please hint me a way to reuse it's logic from ev threads?

Reply:
Here are some draft ideas:
* use some queue that is able to push data to send in ev threads and pop in coro threads to send via non-modified Socket
* add some Socket function with a very long and ugly name that instead of suspending a coroutine immediately returns with some kind of EWOULDBLOCK flag
* refactor the logic to not use libev directly (no idea how to do it in practice, but may be you could came up with some insight)

Reply:
> Could you please hint me a way to reuse it's logic from ev threads?

The easiest way is to not rely on libev availability, it is not really a hard dependency of userver and not a part of public interface.

I had not yet the time to dive into the connection logic but judging by the quick look at the AMPQ-CPP's readme it should not be impossible to just use a Socket/TlsWrapper in some Task buried inside the connection object.

Reply:
Think more about this, i came to conclusion that relying that much on libev and trying to work with `Socket/TlsWrapper` from ev-threads was a sign of bad design by me, and i'm changing the way driver communicates with RabbitMQ completely.

Sorry for the inconvenience and thank you guys for the valuable feedback on this